esmvalcore/preprocessor/_multimodel.py

```diff
@@ -412,14 +433,69 @@ def multi_model_statistics(products,
         )
     if all(type(p).__name__ == 'PreprocessorFile' for p in products):
         # Avoid circular input: https://stackoverflow.com/q/16964467
-        return _multiproduct_statistics(
-            products=products,
-            statistics=statistics,
-            output_products=output_products,
-            span=span,
-            keep_input_datasets=keep_input_datasets,
-        )
+        statistics_products = set()
+        for group, input_prods in _group_products(products, by_key=groupby):
+            sub_output_products = output_products[group]
+
+            # Compute statistics on a single group
+            group_statistics = _multiproduct_statistics(
+                products=input_prods,
+                statistics=statistics,
+                output_products=sub_output_products,
+                span=span,
+                keep_input_datasets=keep_input_datasets
+            )
+
+            statistics_products |= group_statistics
+
+        return statistics_products
     raise ValueError(
         "Input type for multi_model_statistics not understood. Expected "
         "iris.cube.Cube or esmvalcore.preprocessor.PreprocessorFile, "
         "got {}".format(products))
+
+
+def ensemble_statistics(products, statistics,
+                        output_products, keep_input_datasets=True):
+    """Entry point for ensemble statistics.
+
+    An ensemble grouping is performed on the input products.
+    The statistics are then computed calling
+    the :func:`esmvalcore.preprocessor.multi_model_statistics` module,
+    taking the grouped products as an input.
+
+    Parameters
+    ----------
+    products: list
+        Cubes (or products) over which the statistics will be computed.
+    statistics: list
+        Statistical metrics to be computed, e.g. [``mean``, ``max``]. Choose
+        from the operators listed in the iris.analysis package. Percentiles can
+        be specified like ``pXX.YY``.
+    output_products: dict
+        For internal use only. A dict with statistics names as keys and
+        preprocessorfiles as values. If products are passed as input, the
+        statistics cubes will be assigned to these output products.
+    keep_input_datasets: bool
+        If True, the output will include the input datasets.
+        If False, only the computed statistics will be returned.
+
+    Returns
+    -------
+    set
+        A set of output_products with the resulting ensemble statistics.
+
+    See Also
+    --------
+    :func:`esmvalcore.preprocessor.multi_model_statistics` for
+    the full description of the core statistics function.
+    """
+    ensemble_grouping = ('project', 'dataset', 'exp', 'sub_experiment')
+    return multi_model_statistics(
+        products=products,
+        span='overlap',
```

Comment:
All ensemble members for a single dataset I have seen span for the same timerange, so I guess that is why it's fine to assume that they always overlap. But I added the option to keep it flexible, it can always be changed back anyway.